rustler_codegen/src/encode_decode_templates.rs

```diff
@@ -69,7 +133,56 @@ pub(crate) fn decoder(ctx: &Context, inner: TokenStream) -> TokenStream {
 
 pub(crate) fn encoder(ctx: &Context, inner: TokenStream) -> TokenStream {
     let ident = ctx.ident;
-    let generics = ctx.generics;
+    let mut generics = ctx.generics.clone();
+    let type_parameters: Vec<_> = generics
```

Comment:
This is the same filter as further above, we should extract that into a function to avoid duplication.

Reply:
Into  a function or rather a field of `Context`?

Reply:
I guess a field is better :)

Reply:
I'd refactor the lifetimes as well in this case.